platform/pulpcore/app/urls.py

```diff
@@ -1,73 +1,92 @@
 """pulp URL Configuration"""
-
-import warnings
-
-from django.core.exceptions import AppRegistryNotReady
 from django.conf.urls import url, include
 from rest_framework_nested import routers
 
 from pulpcore.app.apps import pulp_plugin_configs
 from pulpcore.app.views import ContentView, StatusView
 
-root_router = routers.DefaultRouter(
-    schema_title='Pulp API',
-    schema_url='/api/v3'
-)  #: The Pulp Platform v3 API router, which can be used to manually register ViewSets with the API.
-
-# Load non-nested viewsets first, so we can make nested routers with them.
-for app_config in pulp_plugin_configs():
-    for viewset in app_config.named_viewsets.values():
-        if not viewset.nest_prefix:
-            viewset.register_with(root_router)
-
-
-nested_routers = (
-    routers.NestedDefaultRouter(root_router, 'repositories', lookup='repository'),
-)
+import logging
+log = logging.getLogger(__name__)
 
 
-def router_for_nested_viewset(viewset):
+class VSNode(object):
     """
-    Find the router for a nested viewset.
-
-    Args:
-        viewset (pulpcore.app.viewsets.NamedModelViewSet): a viewset for which a nested router
-            should exist
-
-    Returns:
-        routers.NestedDefaultRouter: the nested router whose parent_prefix corresponds to the
-            viewset's nest_prefix attribute.
+    This is a tree that can register nested ViewSets with DRF routers.
+    """
+    def __init__(self, viewset=None):
+        self.vs = viewset
+        self.children = []
+
+    def add_node(self, node):
+        """
+        Add a VSNode to the tree.
+
+        Some nodes must be added more than once. In the case of a Master/Detail parent,
+        (example Distributions) the ViewSet's parent_viewset is a master viewset. Each of the detail
+        viewsets belonging will have its own router, and the Distribution viewset must be registered
+        with each of them.
+        """
+        # Master viewsets cannot be registered
+        if node.vs.is_master_viewset():
+            return
+        # Non-nested viewsets are attached to the root node
+        if not node.vs.parent_viewset:
+            self.children.append(node)
+        # link to all parent viewsets
+        elif self.vs and issubclass(self.vs, node.vs.parent_viewset):
+            self.children.append(node)
+        else:
+            for child in self.children:
+                child.add_node(node)
+
+    def register_with(self, router, all_routers=None):
```

Comment:
Suggest:
```
def register_with(self, *router):
```

can be called with:
```
node.register_with(router)
node.register_with(routerA, routerB, ...)
node.register_with(*list_of_router)
```

Reply:
I updated the docstring, because this is a little confusing. The node is registered with `router` only, `all_routers` is just keeping track of all the new routers we create.